libs/wordpress-plugin/plugin/trunk/server/block-db-table.php

```diff
@@ -75,21 +83,35 @@ function block_protocol_set_migration_version_to(int $migration_version)
   }
 }
 
-function block_protocol_migrate()
+function block_protocol_migrations()
 {
-  $saved_version = (int) get_site_option('block_protocol_db_migration_version');
+  return [
+    'block_protocol_migration_1',
+    'block_protocol_migration_2',
+  ];
+}
 
+function block_protocol_migrate()
+{
   // Don't apply migrations if the DB version is unsupported.
   if (!block_protocol_is_database_supported()) {
     return;
   }
 
-  if ($saved_version < 2) {
-    block_protocol_migration_1();
-    block_protocol_set_migration_version_to(2);
-  }
+  $saved_version = ((int) get_site_option('block_protocol_db_migration_version')) ?? 0;
 
-  // future migrations go here
+  $migrations = block_protocol_migrations();
+
+  // Apply new migrations sequentially, skipping already-applied ones.
+  foreach ($migrations as $index => $migration) {
+    // Fetch the saved version at each iteration as it should be changing
+    $saved_version = (int) get_site_option('block_protocol_db_migration_version');
+    $expected_migration_version = $index + 1;
+    if ($saved_version <= $expected_migration_version) {
+      ($migration)();
+      block_protocol_set_migration_version_to($expected_migration_version + 1);
+    }
+  }
```

Comment:
The main change is that we previously checked for `block_protocol_db_migration_version`, which would be `null` in the beginning (cast to 0 I believe) and the set the migration level to `2` after applying migration `1`.

Now we use the same numbering scheme but calculate the current/next migration versions automatically from the array indices.
We have
`[ 0 => "block_protocol_migration_1", 1 => "block_protocol_migration_1"]`
which is translated to 
`[ 1 => "block_protocol_migration_1", 2 => "block_protocol_migration_1"]`
in the loop. We then apply the migration if the current migration is less than or equal to the current loop iteration.
So we'd be applying `1` on a fresh install of the plugin, and on existing installs that have migration set to `2` in their DB, we'd apply `2` because of the equality check. `block_protocol_db_migration_version` is always set to `current migration index + 1 + 1` which would mean we set 
`"block_protocol_migration_1" to 2, "block_protocol_migration_1" to 3`